sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3406,7 +3406,7 @@ pi_result cuda_piextUSMHostAlloc(void **result_ptr, pi_context context,
   } catch (pi_result error) {
     result = error;
   }
-  assert(reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0);
+  assert((alignment == 0) or (reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0));
```

Comment:
```suggestion
  assert((alignment == 0) || (reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0));
```
"or" confuses a lot of people.

Reply:
Mhm.. OK

Reply:
The primary reason to do this change is avoid build issues. See https://github.com/intel/llvm/pull/1501.